README.md

```diff
@@ -1,14 +1,113 @@
 
-As the maintainer of this project, please make a few updates:
+## Features
 
-- Improving this README.MD file to provide a great experience
-- Updating SUPPORT.MD with content about this project's support experience
-- Understanding the security reporting process in SECURITY.MD
-- Remove this section from the README
+- **Local Azure App Configuration API**: Emulates the Azure App Configuration REST API
+- **Web UI**: Provides a web-based interface for managing configuration settings
+- **Multiple Authentication Methods**: Supports HMAC, Entra ID, and anonymous authentication
+- **Configuration Snapshots**: Supports creating and managing configuration snapshots
+- **Cross-platform**: Runs on Windows, macOS, and Linux
+
+## Prerequisites
+
+Before building and running the Azure App Configuration Emulator, ensure you have the following technologies installed:
+
+### For Building and Running
+- **.NET 8.0 SDK** - Required for building and running the C# application
+  - Download from: https://dotnet.microsoft.com/download/dotnet/8.0
+  - Minimum version: 8.0.100
+- **Node.js and npm** - Required for building the React/TypeScript UI component
+  - Download from: https://nodejs.org/
+  - Minimum Node.js version: 16.x or later
+
+## Build
+
+Follow these steps to build the Azure App Configuration Emulator:
+
+### 1. Clone the Repository
+```bash
+git clone https://github.com/Azure/AppConfiguration-Emulator.git
+cd AppConfiguration-Emulator
+```
+
+### 2. Restore Dependencies
+Restore all .NET package dependencies:
+```bash
+dotnet restore
+```
+
+### 3. Build the Solution
+Build the entire solution including the UI components:
+```bash
+dotnet build
+```
+
+The build process will:
+- Compile all C# projects
+- Install npm dependencies for the UI component
+- Build the React/TypeScript frontend using Vite
+- Generate the production-ready UI assets
+
+## Run
+
+To run the Azure App Configuration Emulator locally:
+
+### Start the Emulator
+```bash
+dotnet run --project src/Azure.AppConfiguration.Emulator.Host/Azure.AppConfiguration.Emulator.Host.csproj
+```
+
+### Access the Application
+Once started, the emulator will be available at:
+- **API Endpoint**: `http://127.0.0.1:8483`
+- **Web UI**: `http://127.0.0.1:8483` (serves both API and UI)
+
+### Default Configuration
+The emulator runs with the following default settings in development mode:
+- **Port**: 8483
+- **Authentication**: Anonymous authentication enabled
+- **Anonymous User Role**: Owner (full permissions)
+- **Logging Level**: Debug
+
+### Stopping the Emulator
+Press `Ctrl+C` in the terminal to stop the emulator.
+
+## Test
+
+### Running Tests
+Currently, there are no automated tests included in this solution. To run tests (when available):
```

Comment:
It's fine to call out that there are no automated tests at the moment. but don't call out a future command (dotnet test) that doesn't exist at this time.